core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/CollectorService.scala

```diff
@@ -161,8 +164,9 @@ class CollectorService(
           Failure.GenericFailure(Instant.now(), NonEmptyList.one(error.getMessage)),
           Payload.RawPayload(queryString.getOrElse(""))
         )
-        // We cannot return 400 to the JS tracker, as that will lead to the event getting eternally stuck in the client cache.
-        (HttpResponse(StatusCodes.OK), sinkBad(badRow, partitionKey))
+
+        if (sinks.bad.isHealthy) (HttpResponse(StatusCodes.OK), sinkBad(badRow, partitionKey))
+        else (HttpResponse(StatusCodes.OK), Nil) // if bad sink is unhealthy, we don't want to try storing the bad rows
```

Comment:
Yes, I decided to not revert that change because these bad rows are unrecoverable and there is no point in sinking them, other than to count how many of them there were. But if the sink becomes available very soon, then presumably, we'll only lose a small number of them.

At the same time, shedding them means they won't be trapped in the retry loop, which would give the collector a little bit more time to wait for the sink to recover.